### PR TITLE
feat : use docker buildx to create amd64 images

### PR DESCRIPTION
**Pull request by Bonnai2112**
This branch is the basis for several others in the context of creating arm64 images.
I do this to facilitate proofreading and also to allow gradual merges. The changes made to this branch are:

* Python SRC : Change python docker module from docker to python-on-whales
* Python SRC : Build amd64 images with docker buildx
* Python SRC Config : Add linux/amd64 and linux/arm64 as values for the new config property base_platforms
* CI : use docker/setup-qemu-action@v2 and docker/setup-buildx-action@v1 as gitflows step actions
* Images : Only [OCI images](https://github.com/opencontainers/image-spec/blob/main/spec.md) for amd64 architecture are built

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 Versions
 ========
+2022-07-21
+----------
+**Improvements**
```
Comment (Bladrak) [inline CHANGELOG.md:5]: This can go within the 2022-07-31 section, no need to create a new one :)

File: base_config.yml [hunk 2]
```diff
@@ -6,3 +6,7 @@ DOCKER_TIMEOUT: 600
 # Common build_vars
 MODD_VERSION: &MODD_VERSION 0.5
 TASKFILE_VERSION: &TASKFILE_VERSION 3.2.2
+
+base_platforms: &base_platforms
+  - linux/amd64
```
Comment (Bladrak) [inline base_config.yml:11]: Wouldn't it be easier to have `linux/amd64` as base config and add the arm platform in the images step by step?

**Review by Bladrak — CHANGES_REQUESTED** [review]

**Review by Bladrak — APPROVED** [review]

**Review by Bladrak — APPROVED** [review]

**Review by Bladrak — APPROVED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -47,6 +47,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+      - uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v1
```
Comment (axelpavageau) [inline .github/workflows/build.yml:51]: there's a v2 available.

File: src/config.py [hunk 4]
```diff
@@ -78,3 +78,35 @@ def get_image_fullname(image_name, version, image_conf, env_conf):
...
+    version_tag = f'{version}-' if version != "1" else ""
+
+    if env_conf["tag"]:
+        version_tag += env_conf["tag"] if version != "1" else env_conf["tag"]
+    elif env_conf["event_type"] == "schedule":
+        version_tag += "nightly" if version != "1" else "nightly"
+    elif env_conf["branch"] in ["master"]:
+        version_tag += "latest" if version != "1" else "latest"
+    else:
+        version_tag += "latest" if version != "1" else "latest"
```
Comment (fpavageau) [inline src/config.py:94]: Is this not twice the same thing? I'm reading it as
```python
version_tag += ("latest" if version != "1" else "latest")
```
so it's `"latest"` or `"latest"`? Or is my Python so rusty?

**Review by fpavageau — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (Bonnai2112) [reply .github/workflows/build.yml]: Done.
Commit id : 10d9a80d831fc620f268ba27125cee73fc15dc91

File: ansible/Dockerfile [hunk 5]
```diff
@@ -4,10 +4,10 @@ ARG ANSIBLE_VERSION
...
 
-RUN apt update && apt install -y wget zip && \
+RUN apt update && apt install -y curl zip && \
     pip install -U pip pipenv boto3 && \
     pip install -r requirements.txt &&  \
     rm -rf /var/cache/apk/* && \
-    wget "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" && \
-    unzip awscli-exe-linux-x86_64.zip && \
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
+    unzip awscliv2.zip && \
```
[the remark this answers is not in the document]
Comment (Bonnai2112) [reply ansible/Dockerfile:12]: Done.
Commit id : 0531ce8a6efc40685ca9b1c0b27384c70f040bb4

File: src/config.py (hunk 4 above)
Comment (Bonnai2112) [reply src/config.py:94]: I asked myself the same question, I took over the construction algo of the master branch. I will do other tests to see if we can simplify in the following versions.

Comment (fpavageau) [reply src/config.py:94]: I've found [the refactoring](https://github.com/ekino/docker-buildbox/commit/642e8bee4b3b7f560eedd07bbfc30ee1573ab134#diff-24a3fe5c783755412cb9de83238b4dfd0699345bf50dcadc3ec23c6749361592) which introduced this oddity. It used to be `"-latest" else "latest"`, but the hyphen was moved to the prefix.

So all these constructs can be simplified to `"latest"` (or `"nightly"` or `env_conf["tag"]` depending on the line), both in `get_image_fullname` and `get_image_tags`.

**Review by fpavageau — CHANGES_REQUESTED** [review]

File: src/config.py (hunk 4 above)
Comment (Bonnai2112) [reply src/config.py]: Done !
Commit id : faef7d25b4a3f47cad021c08b4ad614456d423f9

**Review by Bladrak — APPROVED** [review]

**Review by fpavageau — APPROVED** [review]

**Pull request merged**